metricbeat/module/redis/info/data.go

```diff
@@ -237,12 +237,14 @@ var (
 )
 
 // Map data to MapStr
-func eventMapping(info map[string]string) common.MapStr {
+func eventMapping(r mb.ReporterV2, info map[string]string) {
 	// Full mapping from info
 	source := map[string]interface{}{}
 	for key, val := range info {
 		source[key] = val
 	}
 	data, _ := schema.Apply(source)
-	return data
+	r.Event(mb.Event{
```

Comment:
According to this https://github.com/elastic/beats/blob/master/metricbeat/mb/mb.go#L177 `r.Event` returns a `bool` that should be handled or events could not be published at all silently.

Reply:
Awesome, I didn't know about this :smile: I think we are ignoring it everywhere.

But in any case I think that ignoring it is ok. According to the comment you mention this can fail only if the metricset is being stopped, in that case I think this is fine if we just ignore this.

We could think about logging events sent when the metricset is being stopped, but if we do it we should think about how to do it along the current metricsets and I think this would be out of the scope of this PR.

Reply:
I think this requires further discussion, I don't think it is so valuable to log (even at the debug level) that an stopped metricset is failing to send events, this is expected and desired behaviour. If in any case we want to log this on every metricset, then we should do it in the framework itself.

But thinking about this, checking for `false` can be useful to interrupt a `Fetch` that can be collecting multiple events. If one event fails, all the rest would fail too, and it doesn't make sense to continue doing any job there. I explore this option in 344b673, let me know what you think.

Notice that the debug messages added inform about the fetch being interrupted.

Reply:
I think I agree here with @jsoriano As soon as we triggered `Event` it is basically up to the pipeline to decide what to do with the event and we continue. 

Why would all fail if 1 fails?

Reply:
> Why would all fail if 1 fails?

According to documenation (and implementation), `Event()` only returns `false` if the metricset has been stopped, if it is stopped then it will continue also being stopped in subsequent calls and `Event()` will continue without publishing events and returning `false`.

After thinking on this I think it can be the main use of this return value.

Reply:
got it. make sense.